src/services/creatorStaking/creatorsSpaces.ts

```diff
@@ -0,0 +1,51 @@
+import { gql } from 'graphql-request'
+import Cache from '../../cache'
+import { ONE_HOUR } from '../../constant'
+import { soonsocialGraphQlClient } from '../../constant/graphQlClients'
+
+const creatorsSpacesCache = new Cache('creators-spaces', ONE_HOUR)
+
+type CreatorsSpacesInfo = {
+  spaceIds: string[]
+}
+
+export const GET_CREATORS_SPACES = gql`
+  query getCreatorsSpaces($ids: [String!]) {
+    spaces(where: { id_in: $ids, hidden_eq: false }) {
+      id
+      image
+      hidden
+      name
+      about
+      email
+      linksOriginal
+      ownedByAccount {
+        id
+      }
+    }
+  }
+`
+
+export const getCreatorsSpacesInfo = async ({ spaceIds }: CreatorsSpacesInfo) => {
+  const result = await soonsocialGraphQlClient.request(GET_CREATORS_SPACES, { ids: spaceIds })
+
+  if (!result) return []
+
+  const needUpdate = creatorsSpacesCache.needUpdate
+
+  const forceUpdate = needUpdate && (await needUpdate())
+  const cacheData = await creatorsSpacesCache.get()
```

Comment:
afaik, for redis, if you set expiry of any item, it will automatically delete the data,
so isn't it possible to just get() and you don't need to check needUpdate() everytime?

Reply:
> it will automatically delete the data...
 ------------------------------------
 So that why I don't use default redis expiry. I created custom functionality for expire (you can check `src/cache/index.ts` file)